### PR TITLE
Refactor method `wait_for_consensus`

### PR DESCRIPTION
**Pull request by jbajic**
### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: lib/storage/src/content_manager/consensus_manager.rs [hunk 1]
```diff
@@ -600,10 +600,27 @@ impl<C: CollectionContainer> ConsensusManager<C> {
...
+        commit: u64,
+        term: u64,
+        consensus_tick: Duration,
+    ) -> Result<(), ()> {
+        let mut interval = tokio::time::interval(consensus_tick);
 
-        // TODO: naive approach with spinlock for waiting on commit/term, find better way
-        while start.elapsed() < timeout {
+        loop {
+            interval.tick().await;
```
Comment (timvisee) [inline lib/storage/src/content_manager/consensus_manager.rs:623]: We don't want to wait on the first run.

**Review by timvisee — COMMENTED** [review]
Hi! Could you elaborate on what these changes intend to do?

File: lib/storage/src/content_manager/consensus_manager.rs (hunk 1 above)
Comment (jbajic) [reply lib/storage/src/content_manager/consensus_manager.rs:623]: The first tick is triggered immediately as per docs (https://docs.rs/tokio/latest/tokio/time/struct.Interval.html).

**Comment by jbajic** [comment]
> Hi! Could you elaborate on what these changes intend to do?
> 
> We prefer not to touch functions like this unless strictly necessary.

I wanted to resolve TODO comment. My idea was to use a more idiomatic approach. Since tokio supports the exact API functions for this use case.

**Comment by timvisee** [comment]
> I wanted to resolve TODO comment. My idea was to use a more idiomatic approach. Since tokio supports the exact API functions for this use case.

Right!

The problem is, that it does not resolve the TODO. There is still a lock spinning each consensus tick.

Instead this TODO aims to implement it in a different way, probably with some event listening. But we don't have the infrastructure for this in place. And the question is whether we ever will.

**Comment by jbajic** [comment]
> > I wanted to resolve TODO comment. My idea was to use a more idiomatic approach. Since tokio supports the exact API functions for this use case.
> 
> Right!
> 
> The problem is, that it does not resolve the TODO. There is still a lock spinning each consensus tick.
> 
> Instead this TODO aims to implement it in a different way, probably with some event listening. But we don't have the infrastructure for this in place. And the question is whether we ever will.

Aha, that makes sense. I misunderstood the TODO comment, I will close the PR

**Comment by timvisee** [comment]
No problem! Thank you for taking the effort. :smile:

**Pull request closed without merging**